polkadot-protocol/smart-contract-basics/polkavm-design.md

```diff
@@ -42,21 +82,27 @@ PolkaVM introduces two fundamental architectural differences compared to the Eth
     - Reduces overhead for operations not requiring extended precision
     - Facilitates efficient integration with modern CPU architectures
 
-## Pallet Revive
-
-[**`pallet_revive`**](https://paritytech.github.io/polkadot-sdk/master/pallet_revive/index.html){target=\_blank} is a runtime module that executes smart contracts by adding extrinsics, runtime APIs, and logic to convert Ethereum-style transactions into formats compatible with Polkadot SDK-based blockchains. It processes Ethereum-style transactions through the following workflow:
-
-1. Users interact with a proxy server that emulates the Ethereum JSON RPC interface
-2. The proxy server transforms Ethereum transactions into special dispatchable transactions while preserving the original payload
-3. The pallet's internal logic decodes and processes these transactions for blockchain compatibility
-
-This proxy-based approach eliminates the need for node binary modifications, maintaining compatibility across different client implementations. Preserving the original Ethereum transaction payload simplifies adapting existing tools, which can continue processing familiar transaction formats.
-
-## Revive Compiler
-
-[**Revive**](https://github.com/paritytech/revive){target=\_blank} enables Solidity execution on PolkaVM through a particular compilation pipeline:
-
-- Utilizes the standard Solidity compiler (solc) as its foundation
-- Converts Solidity's [YUL](https://docs.soliditylang.org/en/latest/yul.html){target=\_blank} intermediate representation to RISC-V instructions
-- Maintains complete compatibility with all Solidity versions and features
-- Simplifies implementation by focusing on YUL transformation rather than full compiler development
+## Compilation Process
+
+When compiling a Solidity smart contract, the code passes through the following stages:
+
+```mermaid
+flowchart LR
+    Dev[Developer] --> |Solidity\nSource\nCode| Solc
+    
+    subgraph "Compilation Process"
+        direction LR
+        Solc[solc] --> |YUL\nIR| Revive
+        Revive[Revive Compiler] --> |LLVM\nIR| LLVM
+        LLVM[LLVM\nOptimizer] --> |RISC-V ELF\nShared Object| PVMLinker
+    end
+    
+    PVMLinker[PVM Linker] --> PVM[PVM Blob\nwith Metadata]
+```
+
+The compilation process integrates several specialized components:
+
+1. **Solc** - the standard Ethereum Solidity compiler that translates Solidity source code to YUL IR
```

Comment:
```suggestion
1. **Solc** - the standard Ethereum Solidity compiler that translates Solidity source code to [YUL IR](https://docs.soliditylang.org/en/latest/yul.html){target=\_blank}
```
We can link a few external documents for those who aren't as familiar with some of these things or want to understand them more in-depth or compare them to the Pdot things they already know, etc. If it seems like too much, I'll defer to you guys but I would have found this helpful on my first read through.